src/models/Settings.php

```diff
@@ -40,191 +45,317 @@ class Settings extends Model
+     *
+     * See [craft\helpers\ConfigHelper::durationInSeconds()](craft3:craft\helpers\ConfigHelper::durationInSeconds()) for a list of supported value types.
+     * @group Cart
      */
-    public $pdfAllowRemoteImages = false;
+    public $purgeInactiveCartsDuration = 7776000;
 
     /**
-     * @var string The order reference format
+     * @var bool Whether a shipping address is required before making payment on an order.
+     * @group Orders
      */
-    public $orderReferenceFormat = '{{number[:7]}}';
+    public $requireShippingAddressAtCheckout = false;
 
     /**
-     * @var string Default view for Commerce in the CP
-     * @since 2.2
+     * @var bool Whether a billing address is required before making payment on an order.
+     * @group Orders
      */
-    public $defaultView = 'commerce/orders';
+    public $requireBillingAddressAtCheckout = false;
 
     /**
-     * @var string
+     * @var bool Whether shipping method selection is required before making payment on an order.
+     * @group Orders
      */
-    public $cartVariable = 'cart';
+    public $requireShippingMethodSelectionAtCheckout = false;
 
     /**
-     * @var array
+     * @var bool Whether the [customer info tab](customers.md#user-customer-info-tab) should be shown when viewing users in the control panel.
+     * @group General
+     * @since 3.0
      */
-    public $gatewaySettings = [];
+    public $showCustomerInfoTab = true;
 
     /**
-     * @var string
+     * @var string URL for a user to resolve billing issues with their subscription.
+     *
+     * ::: tip
+     * The example templates include [a template for this page](https://github.com/craftcms/commerce/tree/master/example-templates/shop/plans/update-billing-details.twig).
+     * :::
+     * @group Orders
      */
     public $updateBillingDetailsUrl = '';
 
     /**
-     * @var bool
-     * @since 3.0
+     * @var bool Whether the search index for a cart should be updated when saving the cart via `commerce/cart/*` controller actions.
+     *
+     * May be set to `false` to reduce performance impact on high-traffic sites.
+     *
+     * ::: warning
+     * Setting this to `false` will result in fewer index update queue jobs, but you’ll need to manually re-index orders to ensure up-to-date cart search results in the control panel.
+     * :::
+     * @group Cart
+     * @since 3.1.5
      */
-    public $showCustomerInfoTab = true;
+    public $updateCartSearchIndexes = true;
 
     /**
-     * @var bool
-     * @since 3.0.12
+     * @var bool Whether taxes should be calculated based on the billing address instead of the shipping address.
+     * @group Orders
      */
-    public $validateCartCustomFieldsOnSubmission = false;
+    public $useBillingAddressForTax = false;
 
     /**
-     * @var string|null the uri to redirect to after using the load cart url
-     * @since 3.1
+     * @var bool Whether business tax ID should be validated separately from VAT ID.
+     *
```

Comment:
@mattstein this setting is just deciding to add input validation to the address attribute `businessTaxId`.

If you have a vat-based tax rate, it will look at the addresses `businessTaxId` to figure out if the ID is a valid VAT ID from an API and not apply the VAT tax as an adjustment. But this isn't going to stop someone from putting anything they want into the `businessTaxId` field on the address. 

Some people wanted an address validation error when the person put in a bad vat ID. let me know if that makes sense.

Reply:
Thanks @lukeholder. If I’m understanding correctly:

- `false` means Commerce won’t validate `businessTaxId`, which is ideal if I’m not dealing with VAT, not collecting a `businessTaxId`, or handling VAT with my own custom logic (?).
- `true` means Commerce will make sure `businessTaxId` is a valid VAT ID or throw a validation error.

>  But this isn't going to stop someone from putting anything they want into the businessTaxId field on the address.

You mean when it’s `false`? Because it looks like if this was `true` and I entered `handsomeluke` as my `businessTaxId`, the [resulting validation error](https://github.com/craftcms/commerce/blob/da7b828f13beb74a29c015c707ff71d6d91cda02/src/models/Address.php#L392-L392) would require me to provide a valid ID.

> if the ID is a valid VAT ID from an API and not apply the VAT tax as an adjustment

Pardon my ignorance, but if the store collects VAT and I (the customer) provide a VAT ID, any VAT charges on my order will be removed? Presumably because my ID will be used to later determine what, if anything, I owed for tax on that order?